lib/steps.js

```diff
@@ -52,20 +52,36 @@ let click_with = {
 };
 
 BeforeAll(async () => {
+  scope.timeout = 120 * 1000;
+  scope.showif_timeout = 500;
   scope.report = {};
-
 });
 
 Before(async (scenario) => {
-  scope.timeout = 120 * 1000;
-  scope.showif_timeout = 500;
-  scope.scenario_name = scenario.pickle.name.replace(/[^A-Za-z0-9]/gi, '');
+  // Names
+  let scenario_name = scenario.pickle.name.replace(/[^A-Za-z0-9]/gi, '');
+  let tags = ''
+  for ( let tag of scenario.pickle.tags ) {
+    tags += tag.name.replace(/[^A-Za-z0-9]+/g, '_').replace(/(\P{L})+/gu, '_');
+  }
+  // With tag names automatically added to langauges, this should work to make
+  // each language unique.
+  scope.safe_id = `_${ scenario_name }${ tags }`
+
+  // Should we add a counter? It would have to be unique to each scenario name.
+  if ( scope.report[ scope.safe_id ] ) {
+    scope.safe_id += `_1`;
+  }
+
+  // Report
+  scope.report[ scope.safe_id ] = { ids: [] };
```

Comment:
Why not let `scope.addToReport` handle this initialization? Doing it here seems to duplicate work.

Reply:
I like it. Maybe we can group the changes we decide on so that we don't have to test this a million times. Or maybe they should be separate. They're not really related... Guess I'm just getting lazy in my old age.